rules_building_block/credential_access_mdmp_file_creation.toml

```diff
@@ -1,73 +1,82 @@
+bypass_bbr_timing = true
 creation_date = "2023/09/21"
 integration = ["endpoint"]
 maturity = "production"
 min_stack_comments = "New fields added: required_fields, related_integrations, setup"
 min_stack_version = "8.3.0"
-updated_date = "2024/01/10"
-bypass_bbr_timing = true
+updated_date = "2024/03/03"
 
 [rule]
 author = ["Elastic"]
+building_block_type = "default"
 description = """
 Identifies the creation or modification of a medium size memory dump file which can indicate an attempt to access
 credentials from a process memory.
 """
 from = "now-9m"
-index = ["logs-endpoint.events.*"]
+index = ["logs-endpoint.events.file*"]
 language = "eql"
 license = "Elastic License v2"
 name = "Potential Credential Access via Memory Dump File Creation"
+note = "This rule may have low to medium performance impact as it searches for file creation events in general while attempting to ignore specific system paths and known system processes."
 risk_score = 21
 rule_id = "e707a7be-cc52-41ac-8ab3-d34b38c20005"
 severity = "low"
-tags = ["Domain: Endpoint", "OS: Windows", "Use Case: Threat Detection", "Tactic: Credential Access", "Data Source: Elastic Defend", "Rule Type: BBR"]
+tags = [
+    "Domain: Endpoint",
+    "OS: Windows",
+    "Use Case: Threat Detection",
+    "Tactic: Credential Access",
+    "Data Source: Elastic Defend",
+    "Rule Type: BBR",
+]
 timestamp_override = "event.ingested"
-building_block_type = "default"
 type = "eql"
 
 query = '''
 file where host.os.type == "windows" and event.type == "creation" and
 
   /* MDMP header */
-  file.Ext.header_bytes : "4d444d50*" and file.size >= 30000 and
-  not
+  startsWith(file.Ext.header_bytes,  "4d444d50") and
 
+  /* Exclude System process and non-signed executables or exclude events where executables are missing */
+  not process.name == "System" and not ((process.code_signature.trusted == true or process.executable == null)) and
+  not process.code_signature.subject_name : ("Senior Sistemas S/A") and not
   (
     (
-      process.name : "System" or
-      process.executable : (
-        "?:\\Windows\\System32\\WerFault.exe",
-        "?:\\Windows\\SysWOW64\\WerFault.exe",
-        "?:\\Windows\\System32\\Wermgr.exe",
-        "?:\\Windows\\SysWOW64\\Wermgr.exe",
-        "?:\\Windows\\System32\\WerFaultSecure.exe",
-        "?:\\Windows\\SysWOW64\\WerFaultSecure.exe",
-        "?:\\Windows\\System32\\WUDFHost.exe",
-        "C:\\Windows\\System32\\rdrleakdiag.exe",
-        "?:\\Windows\\System32\\Taskmgr.exe",
-        "?:\\Windows\\SysWOW64\\Taskmgr.exe",
+      /* Exclude benign executables */
+      endsWith(process.executable, "WerFault.exe") or
+      endsWith(process.executable, "Wermgr.exe") or
+      endsWith(process.executable, "WerFaultSecure.exe") or
+      endsWith(process.executable, "WUDFHost.exe") or
+      endsWith(process.executable, "rdrleakdiag.exe") or
+      endsWith(process.executable, "Taskmgr.exe") or
+      endsWith(process.executable, "zCrashReport64.exe") or
```

Comment:
This is for sure more expensive, the above would avoid the wildcard condition

Reply:
Yea, after I reviewed it and though about additional benign exceptions that we may add over time, it seemed to be expensive, not to mention the trailing wildcard term searches it would have to do. Reverted.